src/WebLinks/AzureDevOpsWebProvider.cs

```diff
@@ -1,33 +1,23 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.Text.RegularExpressions;
 
 namespace Mjcheetham.WebLinks
 {
     internal class AzureDevOpsWebProvider : IWebProvider
     {
         public string Name => "Azure DevOps";
+        private string SshUriPattern => @"\w+@vs-ssh.visualstudio.com:\w+\/(?<organization>\w+)\/(?<project>\w+)\/(?<repo>\w+)$";
```

Comment:
Unfortunately the possible valid SSH URLs for Azure DevOps is quite large – this regular expression does not capture all possible 'v3' URLs (those not parsable by `System.Uri`).

See your regex test run here: https://regex101.com/r/0rvCdA/1

The following are the possible 'v3' URLs structures:
```
git@dev.azure.com:v3/org/project/repo
git@dev.azure.com:v3/org/project/_full/repo
git@dev.azure.com:v3/org/project/_optimized/repo
git@vs-ssh.visualstudio.com:v3/org/project/repo
git@vs-ssh.visualstudio.com:v3/org/project/_full/repo
git@vs-ssh.visualstudio.com:v3/org/project/_optimized/repo
dev.azure.com:v3/org/project/repo
dev.azure.com:v3/org/project/_full/repo
dev.azure.com:v3/org/project/_optimized/repo
vs-ssh.visualstudio.com:v3/org/project/repo
vs-ssh.visualstudio.com:v3/org/project/_full/repo
vs-ssh.visualstudio.com:v3/org/project/_optimized/repo
```

I would recommend the following regular expression instead ([test run](https://regex101.com/r/RzVEDY/2)):
```
(?:.+@)?(?'host'.+)\:v3\/(?'organization'[^\/]+)\/(?'project'[^\/]+)\/(?:_[^\/]+\/)?(?'repo'[^\/]+)\/?$
```

With this regex you'll need to also extract the host name and check it for *.vs.com and dev.azure.com before proceeding
```csharp
string hostName = match.Groups["host"].Value;
if (StringComparer.OrdinalIgnoreCase.Equals(hostName, "dev.azure.com")
    || hostName.EndsWith(".visualstudio.com", StringComparison.OrdinalIgnoreCase))
{
    // Not Azure DevOps
    return false;
}
```

The good news is that older SSH URLs ('v1' and 'v2') and any 'v3' URLs which include a custom port number will continue to be parsable by the `IsHttpUri` method, so we don't have to worry about those 😄

Reply:
👍